src/v0/destinations/twitter_ads/transform.js

```diff
@@ -131,17 +131,17 @@ function processTrack(message, metadata, destination) {
     identifiers.push({ twclid: message.properties.twclid });
   }
 
-  if (message.properties.ip_address) {
-    const ipAddress = message.properties.ip_address.trim();
+  const ipAddress = message.properties.ip_address?.trim();
+  const userAgent = message.properties.user_agent?.trim();
+
+  if (ipAddress && userAgent) {
+    identifiers.push({ ip_address: ipAddress, user_agent: userAgent });
+  } else if (identifiers.length > 0) {
     if (ipAddress) {
-      identifiers.push({ ip_address: ipAddress });
+      identifiers[0] = { ...identifiers[0], ip_address: ipAddress };
```

Comment:
identifiers[0].ip_address = ipAddress;

Reply:
@krishna2020 this could have been done with Feature flags
cc: @vinayteki95 @dhawal-rudder